testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestSyncPartitionMetadata.java

```diff
@@ -159,6 +161,9 @@ private void prepare(HdfsClient hdfsClient, HdfsDataSourceWriter hdfsDataSourceW
         // should only be picked up when not in case sensitive mode
         hdfsClient.createDirectory(tableLocation + "/COL_X=g/col_y=10");
         hdfsDataSourceWriter.ensureDataOnHdfs(tableLocation + "/COL_X=g/col_y=10", dataSource);
+        // add partition directory with special characters: col_x=h's/col_y=11 with single_int_column/data.orc file
+        hdfsClient.createDirectory(tableLocation + "/col_x=h's/col_y=11");
```

Comment:
should `'` be url encoded here?

Reply:
If it would, it wouldn't allow to reproduce the issue.

Reply:
the path should be created with `FileUtils.makePartName`
it looks like the apostrophe is on the `org.apache.hadoop.hive.common.FileUtils#charToEscape` "list"

Reply:
I've hit this issue when using a table with an external location pointing to an S3 bucket, where I create the partitions and write data outside of Trino. I'm trying to replicate this here.

Should Trino support partitions where values are unescaped?

Reply:
if the data was written "incorrectly" we cannot expect `sync_partition_metadata` to handle that.

so now the question is -- was it written "incorrectly"? not following `FileUtils.makePartName` behavior sounds incorrect, but also -- indistinguishable, since the value could contain escape-like sequences itself

Reply:
Yes. If there's support for arbitrary locations, should sync pick them up too?

Reply:
No. `sync_partition_metadata` can only work with partition location following the "standard", which is "as Hive does"

Reply:
Can you point me to some references? I can add this to docs then. But it needs fixing then anyway, so that it would reject or ignore locations with unescaped values. Right now it tries to add them again so it cannot sync any partitions at all.